src/pveclib/vec_int512_ppc.h

```diff
@@ -0,0 +1,2167 @@
+ *
+ *  \note This structure does not qualify for parameter passing in
+ *  registers (more than 8 registers are required) and will be passed
+ *  in memory.
+ */
+typedef struct
+{
+  ///@cond INTERNAL
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+  vui128_t vx0;
+  vui128_t vx1;
+  vui128_t vx2;
+  vui128_t vx3;
+  vui128_t vx4;
+  vui128_t vx5;
+  vui128_t vx6;
+  vui128_t vx7;
+  vui128_t vx8;
+  vui128_t vx9;
+  vui128_t vx10;
+  vui128_t vx11;
+  vui128_t vx12;
+  vui128_t vx13;
+  vui128_t vx14;
+  vui128_t vx15;
+  vui128_t vx16;
+  vui128_t vx17;
+  vui128_t vx18;
+  vui128_t vx19;
+  vui128_t vx20;
+  vui128_t vx21;
+  vui128_t vx22;
+  vui128_t vx23;
+  vui128_t vx24;
+  vui128_t vx25;
+  vui128_t vx26;
+  vui128_t vx27;
+  vui128_t vx28;
+  vui128_t vx29;
+  vui128_t vx30;
+  vui128_t vx31;
+#else
+  vui128_t vx31;
+  vui128_t vx30;
+  vui128_t vx29;
+  vui128_t vx28;
+  vui128_t vx27;
+  vui128_t vx26;
+  vui128_t vx25;
+  vui128_t vx24;
+  vui128_t vx23;
+  vui128_t vx22;
+  vui128_t vx21;
+  vui128_t vx20;
+  vui128_t vx19;
+  vui128_t vx18;
+  vui128_t vx17;
+  vui128_t vx16;
+  vui128_t vx15;
+  vui128_t vx14;
+  vui128_t vx13;
+  vui128_t vx12;
+  vui128_t vx11;
+  vui128_t vx10;
+  vui128_t vx9;
+  vui128_t vx8;
+  vui128_t vx7;
+  vui128_t vx6;
+  vui128_t vx5;
+  vui128_t vx4;
+  vui128_t vx3;
+  vui128_t vx2;
+  vui128_t vx1;
+  vui128_t vx0;
+#endif
+  ///@endcond
+} __VEC_U_4096;
+
+/*! \brief A vector representation of a 4096-bit unsigned integer
+ * as 8 x 512-bit integers.
+ *
+ *  A Union of:
+ *  - homogeneous aggregate of 4096-bit unsigned integer.
+ *  - struct of:
+ *    - 8 x homogeneous aggregate of 512-bit unsigned integers.
+ *
+ *  \note Useful to access 512-bit blocks to pass to a 512x512-bit
+ *  multiplies. These can be used as partial products in a larger
+ *  2048x2048-bit multiply.
+ *
+ *  \note This structure does not qualify for parameter passing in
+ *  registers (more than 8 registers are required) and will be passed
+ *  in memory.
+ *
+ */
+typedef union
+{
+  ///@cond INTERNAL
+  __VEC_U_4096 x4096;
+  struct
+  {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    __VEC_U_512 v0x512;
+    __VEC_U_512 v1x512;
+    __VEC_U_512 v2x512;
+    __VEC_U_512 v3x512;
+    __VEC_U_512 v4x512;
+    __VEC_U_512 v5x512;
+    __VEC_U_512 v6x512;
+    __VEC_U_512 v7x512;
+#else
+    __VEC_U_512 v7x512;
+    __VEC_U_512 v6x512;
+    __VEC_U_512 v5x512;
+    __VEC_U_512 v4x512;
+    __VEC_U_512 v3x512;
+    __VEC_U_512 v2x512;
+    __VEC_U_512 v1x512;
+    __VEC_U_512 v0x512;
+#endif
+  } x8;
+  ///@endcond
+} __VEC_U_4096x512;
+
+#if 0
+/*! \brief A compiler fence to prevent excessive code motion. */
+#define COMPILE_FENCE __asm (";":::)
+#else
+/*! \brief A compiler fence to prevent excessive code motion. */
+#define COMPILE_FENCE __asm ("nop":::)
+#endif
+
+/*! \brief Macro to add platform suffix for static calls.. */
+#ifdef _ARCH_PWR9
+/*! \brief Macro to add platform suffix for static calls.. */
```

Comment:
Are these meant to be duplicated before the `#ifdef`s start and inside, just before each `#define`? Seems like you either need 1 or 3, not four. :)

Reply:
yea is it not always clean how doxygen will match the \brief description to a conditionally defined macros :-/

But I eliminated that duplicates and it seems come out in the wash.